src/backend/distributed/master/master_repair_shards.c

```diff
@@ -379,7 +390,19 @@ CopyShardForeignConstraintCommandList(ShardInterval *shardInterval)
 		referencedSchemaId = get_rel_namespace(referencedRelationId);
 		referencedSchemaName = get_namespace_name(referencedSchemaId);
 		escapedReferencedSchemaName = quote_literal_cstr(referencedSchemaName);
-		referencedShardId = ColocatedShardIdInRelation(referencedRelationId, shardIndex);
+
```

Comment:
Do we have any other places where we call `ColocatedShardIdInRelation ()`? And, are they dangerous as here in the presence of foreign keys to reference tables?

Reply:
Other than the places we took care of, as I can see, there are some code paths related to Partitioning. Since Postgres does not support foreign keys on partitioned tables, we are not in trouble, yet. Should we dig in and add these checks in case one day we hit such a problem?